### DISCUSSION TITLE
AdminDeleteIdentitySessions with no error whether they are present or not?

### DISCUSSION THREAD

**Discussion by frederikhors** [ory/kratos#2381] [Q&A] [answered]
Currently if I use `V0alpha2Api.AdminDeleteIdentitySessions` it returns `404` if no sessions are found.

I'm wondering if this is useful since that method is used to ensure that all sessions are deleted, whether they are present or not.

Or maybe that's not the meaning of that method and we need another one?

Especially since in Go I get an error when I really don't want it for that case.

```go
httpResponse, err := kratos.V0alpha2Api.AdminDeleteIdentitySessions(ctx, id).Execute()
  if err != nil {
    if httpResponse.StatusCode == 404 {
      return nil
    }

  return err
}
```

**Comment by frederikhors** [comment]
Or maybe we can use Go sentinel errors.

**Comment by aeneasr** [comment, marked as answer]
It would make sense to return 204 in any case!

**Reply by frederikhors** [reply]
Now it is returned `404`, not `204`. In that case (with 204) in Go SDK I shouldn't get error right?

**Reply by aeneasr** [reply]
Yes, exactly, we would need to change the code to return 204 in any case
